### ISSUE TITLE
Stop using RepositoryCache

### ISSUE THREAD

**Issue by jpopelka** [packit/packit-service#1840]
It's been used for SRPM builds which we now (after #1822) do only in `propose-downstream` job and @lachmanfrantisek says that we don't need the repository cache.

* remove the usage from [BaseJobHelper](https://github.com/packit/packit-service/blob/75906e21763e15dc8018ded99aa89c7c892052cf/packit_service/worker/helpers/job_helper.py#L67) & [LocalProjectMixin](https://github.com/packit/packit-service/blob/75906e21763e15dc8018ded99aa89c7c892052cf/packit_service/worker/mixin.py#L183)
* revert packit/deployment#210
* revert packit/packit#1214

@lachmanfrantisek correct me if I'm wrong

**Comment by lachmanfrantisek** [comment]
:+1: 

Maybe to add, the original goal of this was to make it possible to clone kernel (or any other huge repo with a really long history), but they don't use Packit. And we actually don't cache any project if I am checking that correctly.

If we really really want to use that, we might want to check the possibility of read-many volumes to not need to have so many (same) volumes.

**Comment by jpopelka** [comment]
After I resolved #1858 I was checking the logs for some time and saw that the workers spent quite a lot of time cloning `systemd` repo so the cache could still be useful I think.

I guess we can still leave the code in place, close this issue and revisit it in case we'd move to another cluster and decide whether to investigate read-many volumes or disable the cache.

Meanwhile, I reduced (packit/deployment#436) the volumes so that we now request much less storage (all together 72GiB -> 24 GiB if I count correctly).
